errbot/config-template.py

```diff
@@ -93,6 +93,12 @@
 # For example CORE_PLUGINS = ('ACLs', 'Backup', 'Help') you get those names from the .plug files Name entry.
 # For absolutely no plug: CORE_PLUGINS = ()
 
+# Defines an order in which the plugins are getting their callbacks. Useful if you want to have plugins do
+# pre- or post-processing on messages.
+# The 'None' tuple entry represents all the plugins that aren't to be expicitely ordered. For example, if
```

Comment:
Ha this becomes clearer: I would implement that at the backend level,
inheriting from the real backend you want to use and simply override the
incoming message logic there.

On Jun 21, 2016 11:21 AM, "Samuel Lorétan" <notifications@github.com> wrote:

> In errbot/config-template.py
> <https://github.com/errbotio/errbot/pull/791#discussion_r67923527>:
>
> > @@ -93,6 +93,12 @@
> >  # For example CORE_PLUGINS = ('ACLs', 'Backup', 'Help') you get those names from the .plug files Name entry.
> >  # For absolutely no plug: CORE_PLUGINS = ()
> >
> > +# Defines an order in which the plugins are getting their callbacks. Useful if you want to have plugins do
> > +# pre- or post-processing on messages.
> > +# The 'None' tuple entry represents all the plugins that aren't to be expicitely ordered. For example, if
>
> For example. I have a system (separate from the bot) which posts messages
> on a channel on behalf of other users (it does some sort of redirection
> from another separate chat system). All of those messages are posted by a
> "redirector" user, but they all follow the same text structure, with the
> real username in the message.
>
> I want to have an ErrBot plugin which pre-processes messages from the
> redirector to replace the "from" header with the username as parsed in the
> message text, and set the body to the actual content of the message. This
> needs to happen first, so any other plugin will see that message as coming
> from the "real" user, rather than from the redirector.
>
> In that case, I need to enforce a "first this, then everything else"
> order. I would use something like ('RedirectorFixup', None) as
> PLUGINS_CALLBACK_ORDER to express this relationship.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/errbotio/errbot/pull/791/files/5263a0823aee103584e24a2fcbe4839717884851#r67923527>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA7izNL7YEwxiH_C-JxYSMM5uFhogyhQks5qOCuVgaJpZM4I5QzN>
> .
>